src/middleware/index.ts

```diff
@@ -18,14 +35,38 @@ export const logFullRequestMiddleware = (
     `\n` +
       `PATH: ${req.path}\n` +
       `METHOD: ${req.method}\n` +
-      `HEADERS: ${JSON.stringify(scrubTokens(req.headers), null, 2)}\n` +
+      `HOSTNAME: ${req.hostname}\n`,
+    `HEADERS: ${JSON.stringify(scrubTokens(req.headers), null, 2)}\n` +
       `QUERY: ${JSON.stringify(req.query, null, 2)}\n` +
       `BODY: ${JSON.stringify(req.body, null, 2)}\n` +
-      `URL: ${req.url}`
+      `URL: ${req.url}\n` +
+      `ORIGINAL_URL: ${req.originalUrl}\n` +
+      `BASE_URL: ${req.baseUrl}\n` +
+      `PARAMS: ${JSON.stringify(req.params, null, 2)}`
   );
   next();
 };
 
+export const cacheMiddleware = (
+  req: Request,
+  _res: Response,
+  next: NextFunction
+) => {
+  req.cache = {
+    providers: cachedProviders,
+    cloudProviders: cachedCloudProviders,
+  };
+
+  if (process.env.IS_LOCAL === "true") {
```

Comment:
when running locally and running tests after each request the caches are cleared